sky/data/storage.py

```diff
@@ -214,25 +286,31 @@ def __init__(self,
                      name: str,
                      source: Optional[SourceType],
                      region: Optional[str] = None,
-                     is_sky_managed: Optional[bool] = None):
+                     is_sky_managed: Optional[bool] = None,
+                     _bucket_sub_path: Optional[str] = None):
             self.name = name
             self.source = source
             self.region = region
             self.is_sky_managed = is_sky_managed
+            self._bucket_sub_path = _bucket_sub_path
 
         def __repr__(self):
             return (f'StoreMetadata('
                     f'\n\tname={self.name},'
                     f'\n\tsource={self.source},'
                     f'\n\tregion={self.region},'
-                    f'\n\tis_sky_managed={self.is_sky_managed})')
-
-    def __init__(self,
-                 name: str,
-                 source: Optional[SourceType],
-                 region: Optional[str] = None,
-                 is_sky_managed: Optional[bool] = None,
-                 sync_on_reconstruction: Optional[bool] = True):
+                    f'\n\tis_sky_managed={self.is_sky_managed},'
+                    f'\n\t_bucket_sub_path={self._bucket_sub_path})')
+
+    def __init__(
+        self,
+        name: str,
+        source: Optional[SourceType],
+        region: Optional[str] = None,
+        is_sky_managed: Optional[bool] = None,
+        sync_on_reconstruction: Optional[bool] = True,
+        _bucket_sub_path: Optional[str] = None,  # pylint: disable=invalid-name
+        _allow_bucket_creation: bool = True):  # pylint: disable=invalid-name
```

Comment:
Why is this `_allow_bucket_creation ` arg needed? 

Can we instead check store.is_sky_managed field after the store's initialization to check if bucket was created, and if it was created then raise an error? 

Eg. near L727 in controller_utils.py, we can initialize the store object and check is_sky_managed, and raise error appropriately.

```
store = store_cls(name=bucket_name,
                        source=workdir,
                        _bucket_sub_path=bucket_sub_path,
                        **store_kwargs)
if store.is_sky_managed:
  # Bucket was created, this should not happen since the bucket already exists
  # cleanup and raise error
  store.delete()
  raise ...
```

Let me know if I'm missing something. It would be great if we could avoid adding new arguments to the Store/Storage interface.

Reply:
We can do that in `controller_utils.py`. This avoids adding a parameter but keeps the logic outside of the store. If we're not going to reuse the limitation of creating a bucket, it will be fine to handle it in `controller_utils.py`